modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -205,6 +205,35 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
       .forall(e => e == RegistryError.NotFound || e.isInstanceOf[RegistryError.ClientFailure])
   }
 
+  /**
+   * Checks if the resolution error indicates a fatal/unrecoverable issue.
+   *
+   * Returns true when:
+   * - Any custom (non-Iglu Central) repository has a RepoFailure or ClientFailure
+   * - All Iglu Central mirrors have a RepoFailure or ClientFailure
+   *
+   * Returns false otherwise.
+   *
+   * If a registry has a fatal error alongside other errors, assume it is unrecoverable.
+   */
+  def isUnrecoverable(error: ResolutionError): Boolean = {
```

Comment:
I have doubts about the name `isUnrecoverable`.  Bearing in mind this becomes part of the public API of this library, so name is quite important.

If `isUnrecoverable` returns `false` then it implies this error is "recoverable".  But what does "recoverable" mean in this context?  It means we're going to send this event to the bad stream... but I wouldn't say that is an example of "being recoverable".

You know I'm not good at naming things :) so I don't have good suggestions to change the name.

Maybe `isSystemError` or `isPlatformError`... meaning the system/platform/deployment/configuration is the problem, and there is no problem with the data.

Reply:
I think `isSystemError` sounds good, thanks for the suggestion!